File: experiments/020-PythonVsJava/cells/java-gradle_1-Empty_free/2022-1c-recuperatorio-parcial1/20260921-204311-76559/output/project/src/main/java/customerimporter/Environment.java
package customerimporter;

import java.util.Arrays;
import java.util.NoSuchElementException;

public abstract class Environment {

    // current

    public static Environment current() {
        for (Environment anEnvironment : Arrays.asList(new DevelopmentEnvironment(), new IntegrationEnvironment())) {
            if (anEnvironment.isCurrent()) return anEnvironment;
        }
        throw new NoSuchElementException("Object is not in the collection.");
    }

    public abstract boolean isCurrent();

    // customer system

    public abstract CustomerSystem createCustomerSystem();
}

File: experiments/020-PythonVsJava/cells/java-gradle_1-Empty_free/2022-1c-recuperatorio-parcial1/20260921-202259-53404/exercise/python/src/customerimporter/environment.py
from abc import ABC, abstractmethod


class Environment(ABC):

    # current

    @classmethod
    def current(cls):
        for an_environment_class in cls.__subclasses__():
            if an_environment_class.is_current(): return an_environment_class()
        raise RuntimeError("Object is not in the collection.")

    @classmethod
    @abstractmethod
    def is_current(cls):
        pass

    # customer system

    @abstractmethod
    def create_customer_system(self):
        pass
